Doc/library/datetime.rst

```diff
@@ -2538,7 +2538,21 @@ differences between platforms in handling of unsupported format specifiers.
    ``%G``, ``%u`` and ``%V`` were added.
 
 .. versionadded:: 3.12
-   ``%:z`` was added.
+   ``%:z`` was added for :meth:`~.datetime.strftime`
+
+.. versionadded:: 3.13
+   ``%:z`` was added for :meth:`~.datetime.strptime`
+
+.. warning::
```

Comment:
IMHO, this entire warning is unnecessary, because:

* There is no need to *warn*, only to *describe* certain details (see below). After all, the only inconsistency is that ``%z`` is little bit more liberal when it comes to parsing. That's all.
* The statement that *the behavior is unintended* is inaccurate. The behavior is intended and sometimes very useful.
* The recommendation *to use `%z` directive only to parse strings formatted according to `%z` directive, while using ``%:z`` directive for strings formatted according to ``%:z``* is unnecessary. If somebody needs to accept only `:`-based notation, they will parse their input with `%:z`. If somebody needs to be more *liberal* in respect to the input, they will parse their input with `%z`. Both approaches are perfectly OK, so there is no need to discourage people from anything here.

On the other hand, it is worth to *describe* certain details – but, IMHO, a better place to do so is the note `(6)` in the *Technical Detail* subsection (below). Namely, I propose to:

  * (1) Replace the following fragment of the `%z` description:

    ```rst
    For example, ``'+01:00:00'`` will be parsed as an offset of one hour.
    In addition, providing ``'Z'`` is identical to ``'+00:00'``.
    ```

    with the following text:

    ```rst
    For example, both ``'+010000'`` and ``'+01:00:00'`` will be parsed as an offset
    of one hour. In addition, providing ``'Z'`` is identical to ``'+00:00'``.
    ```

  * (2) Replace the following  fragment of the `%:z` description:

    ```rst
    Behaves exactly as ``%z``, but has a colon separator added between
    hours, minutes and seconds.
    ```

    with the following text:

    ```rst
    When used with :meth:`~.datetime.strftime`, behaves exactly as ``%z``, except
    that a colon separator is added between hours, minutes and seconds.

    When used with :meth:`~.datetime.stpftime`, the UTC offsets are *required*
    to have a colon as a separator between hours, minutes and seconds.
    For example, ``'+01:00:00'`` (but *not* ``'+010000'``) will be parsed as an
    offset of one hour. In addition, providing ``'Z'`` is identical to ``'+00:00'``.
    ```